crates/chain/src/indexed_tx_graph.rs

```diff
@@ -29,6 +26,14 @@ impl<A, I: Default> Default for IndexedTxGraph<A, I> {
 }
 
 impl<A, I> IndexedTxGraph<A, I> {
+    /// Construct a new [`IndexedTxGraph`] with a given `index`.
+    pub fn new(index: I) -> Self {
```

Comment:
Why can't you pass in both graph and index here. Just index everything in the graph.

Reply:
Indexing data may result in `I::Additions`. Then the method needs to return both `Self` and `IndexedAdditions`. I don't think this is the right API.

The intention for constructing `IndexedTxGraph` with only an index is so you can set up things like keychains and lookahead settings. I don't think you would ever want to add transactions into `TxGraph` before this stage.

Reply:
Ok understood. I think there should be some way of taking an existing graph and and index and creating an indexed tx graph but we can leave that for another time.